code/modules/mob/living/living_say.dm

```diff
@@ -137,10 +137,14 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 	if(!language)
 		language = get_selected_language()
-
+	var/mob/living/carbon/human/H = src
 	if(!can_speak_vocal(message))
-		to_chat(src, "<span class='warning'>You find yourself unable to speak!</span>")
-		return
+		if (HAS_TRAIT(src, TRAIT_SIGN_LANG) && H.mind.miming)
+			to_chat(src, "<span class='warning'>You stop yourself from signing in favor of the artform of mimery!</span>")
+			return
+		else
+			to_chat(src, "<span class='warning'>You find yourself unable to speak!</span>")
```

Comment:
Any way to do this better/how necessary is this?

Reply:
I didn't believe it necessary, personally, but I've gotten some complaints about the generic chat message for mimes, so I put it in anyways. I was originally going to put it under `/mob/living/proc/can_speak_vocal(message)`, but that would have made both `"You find yourself unable to speak!"` and `"You stop yourself from signing in favor of the artform of mimery!"` to show up at the same time. I will admit the chat message might be a bit long, but I couldn't think of a better way of wording it myself.

Reply:
The was it's placed seems a bit hacky, but I'm not sure there's much to be done.